__init__.py

```diff
@@ -106,40 +113,93 @@ def respond(self, query):
         if len(query) == 0:
             return 0.0
 
-        r = ddg.query(query)
+        # note: '1+1' throws an exception
+        try:
+            r = ddg.query(query)
+        except:
+            LOG.warning("DDG exception!")
+            return None
 
-        LOG.debug('Query: ' + str(query))
-        LOG.debug('Type: ' + r.type)
+        LOG.debug("Query: %s" % (str(query),))
+        LOG.debug("Type: %s" % (r.type,))
+
+        # if disambiguation, save old result
+        # for fallback but try to get the 
+        # real abstract
+        if r.type == 'disambiguation':
+            if r.related:
+                detailed_url = r.related[0].url + "?o=x"
+                LOG.debug("DDG: disambiguating %s" % (detailed_url,))
+                request = requests.get(detailed_url)
+                response = request.content
+                if response:
+                    xml = ElementTree.fromstring(response)
+                    r = ddg.Results(xml)
```

Comment:
avoid one letter variable names

Reply:
I'll let the original author know. I'll also add it to the generic clean up ticket.